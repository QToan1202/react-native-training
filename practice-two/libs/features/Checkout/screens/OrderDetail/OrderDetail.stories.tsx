import { ComponentMeta, ComponentStory } from '@storybook/react-native'

import { OrderDetail } from '../index'

export default {
  title: 'screens/Checkout/Order Detail',
  component: OrderDetail,
} as ComponentMeta<typeof OrderDetail>

const Template: ComponentStory<typeof OrderDetail> = () => <OrderDetail />

export const Default = Template.bind({})
Default.args = {}
