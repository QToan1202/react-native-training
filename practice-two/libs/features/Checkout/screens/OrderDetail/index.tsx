import { useCallback, useMemo } from 'react'
import { Dimensions, Image } from 'react-native'
import { NativeStackScreenProps } from '@react-navigation/native-stack'
import { ScrollView, Separator, Spinner, XStack, YStack } from 'tamagui'

import { Button, Heading, Paragraph } from '@practice-two/shared/components'
import { Feature } from '@practice-two/shared/hocs'
import { ICart, RootStackParamList } from '@practice-two/shared/types'

import { CartItem, TrackerItem } from '../../components'
import { useFindMultiProduct, useGetOderDetail } from '../../hooks'

export type OrderDetailScreenProps = NativeStackScreenProps<RootStackParamList, 'OrderDetail'>

const OrderDetail = ({ navigation, route }: OrderDetailScreenProps) => {
  const { id } = route.params
  const { data, isSuccess } = useGetOderDetail(process.env.ORDER_ENDPOINT, id)
  const products: Partial<ICart>[] = useFindMultiProduct(
    process.env.PRODUCT_ENDPOINT,
    data?.productId || [],
    data?.quantity || []
  )

  const handleNavigateToHome = useCallback(
    () => navigation.navigate('Tabs', { screen: 'HomeTab' }),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    []
  )
  const renderOrderItem = useMemo(() => {
    if (!isSuccess) return null

    return products.map((item: Partial<ICart>) => {
      const { id: itemId, img, name, price, discountPrice, quantity } = item

      if (!itemId || !img || !name || !price || !quantity) return null

      return (
        <CartItem
          key={itemId}
          image={{ uri: img }}
          name={name}
          price={price}
          discountPrice={discountPrice}
          quantity={quantity}
        />
      )
    })
  }, [isSuccess, products])

  return (
    <Feature feat="checkout">
      {isSuccess ? (
        <ScrollView showsVerticalScrollIndicator={false} backgroundColor="$color.bg_layer">
          <YStack alignItems="center" paddingVertical="$space.6">
            <Image source={require('@assets/order/done.png')} />
            <Heading content="Thanks for Order" color="$color.gray_50" />
          </YStack>
          <YStack>{renderOrderItem}</YStack>
          <YStack
            marginTop="$space.3.5"
            marginBottom="$space.5"
            paddingLeft="$space.4.5"
            paddingRight="$space.3.5"
            backgroundColor="$color.white"
          >
            <YStack marginVertical="$space.3" space="$space.2.5">
              <Heading
                content="track order"
                fontWeight="$3"
                lineHeight="$4"
                fontSize="$2"
                color="$color.gray_50"
                textTransform="capitalize"
              />
              <Paragraph
                content="Order ID - 123455"
                fontWeight="$2"
                color="$color.gray_400"
                lineHeight="$3"
              />
            </YStack>
            <Separator borderColor="$color.primary" borderBottomWidth={3} width={70} />
            <YStack marginTop="$space.5" marginBottom="$space.3" space="$space.7">
              <TrackerItem
                trackStatus="order placed"
                description="Order#123455 from Fashion Point"
                date={new Date('2019-05-08')}
                time="11:30 AM"
              />
              <TrackerItem
                trackStatus="payment confirmed"
                description="Payment Confirmed Status"
                date={new Date('2019-05-08')}
                time="11:30 AM"
              />
              <TrackerItem
                trackStatus="processed"
                description="Processed Status"
                date={new Date('2019-05-08')}
                time="11:30 AM"
              />
              <TrackerItem
                trackStatus="delivered"
                description="Delivered Status"
                date={new Date('2019-05-08')}
                time="11:30 AM"
              />
            </YStack>
          </YStack>
          <YStack
            paddingVertical="$space.3"
            paddingHorizontal="$space.4"
            backgroundColor="$color.white"
          >
            <Heading
              content="delivery address"
              fontWeight="$3"
              lineHeight="$4"
              fontSize="$2"
              color="$color.gray_50"
              textTransform="capitalize"
            />
            <Separator
              marginTop="$space.3"
              marginLeft="$space.-3"
              borderColor="$color.divider"
              width={Dimensions.get('window').width}
            />
            <YStack paddingVertical="$space.3" space="$space.2.5">
              <Paragraph
                content="Tradly team"
                fontWeight="$2"
                color="$color.dark_50"
                lineHeight="$3"
              />
              <Paragraph
                content="Flat Number 512, Eden Garden, Rewari"
                fontWeight="$2"
                fontSize={12}
                color="$color.gray_400"
                lineHeight="$3"
              />
              <XStack alignItems="center" space="$space.1.5">
                <Paragraph
                  content="Mobile: "
                  fontWeight="$2"
                  fontSize={12}
                  color="$color.gray_400"
                  lineHeight="$3"
                />
                <Paragraph
                  content="9876543210"
                  fontWeight="$2"
                  fontSize={12}
                  color="$color.dark_50"
                  lineHeight="$3"
                />
              </XStack>
            </YStack>
          </YStack>
          <Button
            variant="quaternary"
            backgroundColor="$color.transparent"
            borderColor="$color.transparent"
            title="Back to Home"
            marginTop="$space.3.5"
            marginBottom="$space.7"
            fontWeight="$3"
            fontSize="$1"
            lineHeight="$4"
            color="$color.dark_50"
            textTransform="none"
            onPress={handleNavigateToHome}
          />
        </ScrollView>
      ) : (
        <Spinner size="large" color="$color.primary" />
      )}
    </Feature>
  )
}

export default OrderDetail
