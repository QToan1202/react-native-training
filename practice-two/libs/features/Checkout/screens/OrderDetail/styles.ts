import { Dimensions, StyleSheet } from 'react-native'

import { COLORS, FONT_FAMILY } from '@constants'

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.BG_LAYER,
  },
  header: {
    alignItems: 'center',
    paddingVertical: 30,
  },
  heading: {
    color: COLORS.GRAY_50,
  },
  detail: {
    marginTop: 16,
    marginBottom: 24,
    paddingLeft: 20,
    paddingRight: 17,
    backgroundColor: COLORS.WHITE,
  },
  titleContainer: {
    marginVertical: 14,
    rowGap: 10,
  },
  title: {
    fontFamily: FONT_FAMILY.MONTSERRAT[600],
    lineHeight: 20,
    fontSize: 16,
    color: COLORS.DARK_50,
    textTransform: 'capitalize',
  },
  text: {
    fontFamily: FONT_FAMILY.MONTSERRAT[500],
    lineHeight: 18,
    color: COLORS.GRAY_400,
  },
  textDark: {
    color: COLORS.DARK_50,
  },
  line: {
    width: 70,
    height: 3,
    backgroundColor: COLORS.PRIMARY,
  },
  status: {
    marginTop: 25,
    marginBottom: 12,
    rowGap: 42,
  },
  addressInfo: {
    backgroundColor: COLORS.WHITE,
    paddingVertical: 14,
    paddingHorizontal: 18,
  },
  divider: {
    marginTop: 12,
    marginLeft: -14,
    borderBottomColor: COLORS.DIVIDER,
    width: Dimensions.get('window').width,
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
  addressContent: {
    paddingVertical: 12,
    rowGap: 9,
  },
  addressText: {
    fontSize: 12,
  },
  btn: {
    backgroundColor: COLORS.TRANSPARENT,
    borderColor: COLORS.TRANSPARENT,
    marginTop: 16,
    marginBottom: 40,
  },
  btnTitle: {
    textTransform: 'none',
    fontFamily: FONT_FAMILY.MONTSERRAT[600],
    lineHeight: 20,
    fontSize: 14,
    color: COLORS.DARK_50,
  },
})

export default styles
