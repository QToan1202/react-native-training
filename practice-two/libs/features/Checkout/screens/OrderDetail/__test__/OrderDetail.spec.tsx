import { render } from '@testing-library/react-native'

import OrderDetail from '../index'

describe('Testing Order Detail screen', () => {
  it('Render screen successfully', () => {
    const component = render(<OrderDetail />).toJSON()

    expect(component).toMatchSnapshot()
  })
})
