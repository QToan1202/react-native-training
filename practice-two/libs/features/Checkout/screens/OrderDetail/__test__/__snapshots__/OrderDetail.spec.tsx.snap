// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Order Detail screen Render screen successfully 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "backgroundColor": "#f6f9ff",
    }
  }
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "paddingVertical": 30,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/order/done.png",
          }
        }
      />
      <Text
        style={
          [
            {
              "color": "#fff",
              "fontFamily": "Montserrat_700Bold",
              "fontSize": 24,
            },
            false,
            {
              "color": "#4f4f4f",
            },
          ]
        }
      >
        Thanks for Order
      </Text>
    </View>
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#fff",
            "opacity": 1,
            "paddingBottom": 12,
            "paddingLeft": 16,
            "paddingTop": 29,
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
              "gap": 15,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/cart/item.png",
              }
            }
            style={
              {
                "borderRadius": 10,
                "height": 102,
                "width": 102,
              }
            }
          />
          <View
            style={
              {
                "justifyContent": "center",
                "rowGap": 12.5,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#4f4f4f",
                    "fontFamily": "Montserrat_500Medium",
                    "textTransform": "capitalize",
                  },
                ]
              }
            >
              coca cola
            </Text>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "columnGap": 4,
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "#fff",
                      "fontFamily": "Montserrat_400Regular",
                    },
                    {
                      "fontSize": 14,
                    },
                    {
                      "color": "#33907C",
                      "fontFamily": "Montserrat_700Bold",
                      "fontSize": 18,
                    },
                  ]
                }
              >
                $25
              </Text>
              <Text
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "#fff",
                      "fontFamily": "Montserrat_400Regular",
                    },
                    {
                      "fontSize": 14,
                    },
                    [
                      {
                        "color": "#4f4f4f",
                        "fontFamily": "Montserrat_500Medium",
                        "textTransform": "capitalize",
                      },
                      {
                        "textDecorationLine": "line-through",
                      },
                    ],
                  ]
                }
              >
                $50
              </Text>
              <Text
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "#fff",
                      "fontFamily": "Montserrat_400Regular",
                    },
                    {
                      "fontSize": 14,
                    },
                    {
                      "color": "#4f4f4f",
                      "fontFamily": "Montserrat_500Medium",
                      "textTransform": "capitalize",
                    },
                  ]
                }
              >
                50% off
              </Text>
            </View>
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#4f4f4f",
                    "fontFamily": "Montserrat_500Medium",
                    "textTransform": "capitalize",
                  },
                ]
              }
            >
              qty: 1
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#fff",
          "marginBottom": 24,
          "marginTop": 16,
          "paddingLeft": 20,
          "paddingRight": 17,
        }
      }
    >
      <View
        style={
          {
            "marginVertical": 14,
            "rowGap": 10,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#fff",
                "fontFamily": "Montserrat_700Bold",
                "fontSize": 24,
              },
              false,
              {
                "color": "#212121",
                "fontFamily": "Montserrat_600SemiBold",
                "fontSize": 16,
                "lineHeight": 20,
                "textTransform": "capitalize",
              },
            ]
          }
        >
          track order
        </Text>
        <Text
          style={
            [
              {
                "color": "#fff",
                "fontFamily": "Montserrat_400Regular",
              },
              {
                "fontSize": 14,
              },
              {
                "color": "#606a7b",
                "fontFamily": "Montserrat_500Medium",
                "lineHeight": 18,
              },
            ]
          }
        >
          Order ID - 123455
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#33907C",
            "height": 3,
            "width": 70,
          }
        }
      />
      <View
        style={
          {
            "marginBottom": 12,
            "marginTop": 25,
            "rowGap": 42,
          }
        }
      >
        <View
          style={
            [
              {
                "rowGap": 6,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "columnGap": 4,
                  "flexDirection": "row",
                },
                {
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  [
                    {
                      "color": "#606a7b",
                      "fontFamily": "Montserrat_500Medium",
                      "fontSize": 10,
                      "lineHeight": 18,
                    },
                    {
                      "color": "#212121",
                      "fontSize": 14,
                      "textTransform": "capitalize",
                    },
                  ],
                ]
              }
            >
              order placed
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#606a7b",
                    "fontFamily": "Montserrat_500Medium",
                    "fontSize": 10,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              08/05/2019
            </Text>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "columnGap": 4,
                  "flexDirection": "row",
                },
                {
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#606a7b",
                    "fontFamily": "Montserrat_500Medium",
                    "fontSize": 10,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              Order#123455 from Fashion Point
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#606a7b",
                    "fontFamily": "Montserrat_500Medium",
                    "fontSize": 10,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              11:30 AM
            </Text>
          </View>
        </View>
        <View
          style={
            [
              {
                "rowGap": 6,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "columnGap": 4,
                  "flexDirection": "row",
                },
                {
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  [
                    {
                      "color": "#606a7b",
                      "fontFamily": "Montserrat_500Medium",
                      "fontSize": 10,
                      "lineHeight": 18,
                    },
                    {
                      "color": "#212121",
                      "fontSize": 14,
                      "textTransform": "capitalize",
                    },
                  ],
                ]
              }
            >
              payment confirmed
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#606a7b",
                    "fontFamily": "Montserrat_500Medium",
                    "fontSize": 10,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              08/05/2019
            </Text>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "columnGap": 4,
                  "flexDirection": "row",
                },
                {
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#606a7b",
                    "fontFamily": "Montserrat_500Medium",
                    "fontSize": 10,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              Payment Confirmed Status
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#606a7b",
                    "fontFamily": "Montserrat_500Medium",
                    "fontSize": 10,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              11:30 AM
            </Text>
          </View>
        </View>
        <View
          style={
            [
              {
                "rowGap": 6,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "columnGap": 4,
                  "flexDirection": "row",
                },
                {
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  [
                    {
                      "color": "#606a7b",
                      "fontFamily": "Montserrat_500Medium",
                      "fontSize": 10,
                      "lineHeight": 18,
                    },
                    {
                      "color": "#212121",
                      "fontSize": 14,
                      "textTransform": "capitalize",
                    },
                  ],
                ]
              }
            >
              processed
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#606a7b",
                    "fontFamily": "Montserrat_500Medium",
                    "fontSize": 10,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              08/05/2019
            </Text>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "columnGap": 4,
                  "flexDirection": "row",
                },
                {
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#606a7b",
                    "fontFamily": "Montserrat_500Medium",
                    "fontSize": 10,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              Processed Status
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#606a7b",
                    "fontFamily": "Montserrat_500Medium",
                    "fontSize": 10,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              11:30 AM
            </Text>
          </View>
        </View>
        <View
          style={
            [
              {
                "rowGap": 6,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "columnGap": 4,
                  "flexDirection": "row",
                },
                {
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  [
                    {
                      "color": "#606a7b",
                      "fontFamily": "Montserrat_500Medium",
                      "fontSize": 10,
                      "lineHeight": 18,
                    },
                    {
                      "color": "#212121",
                      "fontSize": 14,
                      "textTransform": "capitalize",
                    },
                  ],
                ]
              }
            >
              delivered
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#606a7b",
                    "fontFamily": "Montserrat_500Medium",
                    "fontSize": 10,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              08/05/2019
            </Text>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "columnGap": 4,
                  "flexDirection": "row",
                },
                {
                  "justifyContent": "space-between",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#606a7b",
                    "fontFamily": "Montserrat_500Medium",
                    "fontSize": 10,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              Delivered Status
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#606a7b",
                    "fontFamily": "Montserrat_500Medium",
                    "fontSize": 10,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              11:30 AM
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#fff",
          "paddingHorizontal": 18,
          "paddingVertical": 14,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#fff",
              "fontFamily": "Montserrat_700Bold",
              "fontSize": 24,
            },
            false,
            {
              "color": "#212121",
              "fontFamily": "Montserrat_600SemiBold",
              "fontSize": 16,
              "lineHeight": 20,
              "textTransform": "capitalize",
            },
          ]
        }
      >
        delivery address
      </Text>
      <View
        style={
          {
            "borderBottomColor": "#0000001a",
            "borderBottomWidth": 0.5,
            "marginLeft": -14,
            "marginTop": 12,
            "width": 750,
          }
        }
      />
      <View
        style={
          {
            "paddingVertical": 12,
            "rowGap": 9,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#fff",
                "fontFamily": "Montserrat_400Regular",
              },
              {
                "fontSize": 14,
              },
              [
                {
                  "color": "#606a7b",
                  "fontFamily": "Montserrat_500Medium",
                  "lineHeight": 18,
                },
                {
                  "color": "#212121",
                },
              ],
            ]
          }
        >
          Tradly team
        </Text>
        <Text
          style={
            [
              {
                "color": "#fff",
                "fontFamily": "Montserrat_400Regular",
              },
              {
                "fontSize": 14,
              },
              [
                {
                  "color": "#606a7b",
                  "fontFamily": "Montserrat_500Medium",
                  "lineHeight": 18,
                },
                {
                  "fontSize": 12,
                },
              ],
            ]
          }
        >
          Flat Number 512, Eden Garden, Rewari
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "columnGap": 4,
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#fff",
                  "fontFamily": "Montserrat_400Regular",
                },
                {
                  "fontSize": 14,
                },
                [
                  {
                    "color": "#606a7b",
                    "fontFamily": "Montserrat_500Medium",
                    "lineHeight": 18,
                  },
                  {
                    "fontSize": 12,
                  },
                ],
              ]
            }
          >
            Mobile: 
          </Text>
          <Text
            style={
              [
                {
                  "color": "#fff",
                  "fontFamily": "Montserrat_400Regular",
                },
                {
                  "fontSize": 14,
                },
                [
                  {
                    "color": "#606a7b",
                    "fontFamily": "Montserrat_500Medium",
                    "lineHeight": 18,
                  },
                  {
                    "color": "#212121",
                  },
                  {
                    "fontSize": 12,
                  },
                ],
              ]
            }
          >
            9876543210
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 0,
          "borderWidth": 2,
          "marginBottom": 40,
          "marginTop": 16,
          "opacity": 1,
          "paddingVertical": 13,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "columnGap": 6,
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#fff",
                "fontFamily": "Montserrat_400Regular",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "color": "#000",
                },
                {
                  "fontFamily": "Montserrat_500Medium",
                  "textAlign": "center",
                  "textTransform": "capitalize",
                },
                {
                  "color": "#212121",
                  "fontFamily": "Montserrat_600SemiBold",
                  "fontSize": 14,
                  "lineHeight": 20,
                  "textTransform": "none",
                },
              ],
            ]
          }
        >
          Back to Home
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
